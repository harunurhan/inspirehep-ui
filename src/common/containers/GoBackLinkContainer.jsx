import React, { Component } from 'react';
import { connect } from 'react-redux';
import { goBack } from 'react-router-redux';
import PropTypes from 'prop-types';

class GoBackLink extends Component {
  render() {
    const { children, onClick } = this.props;
    return (
      /* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions, jsx-a11y/anchor-is-valid */
      <a onClick={onClick}>{children}</a>
    );
  }
}

GoBackLink.propTypes = {
  onClick: PropTypes.func.isRequired,
  children: PropTypes.string,
};

GoBackLink.defaultProps = {
  children: 'go back',
};

export const dispatchToProps = dispatch => ({
  onClick() {
    dispatch(goBack());
  },
});

export default connect(null, dispatchToProps)(GoBackLink);
