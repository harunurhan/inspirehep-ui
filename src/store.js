import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import createHistory from 'history/createBrowserHistory';
import { routerMiddleware } from 'react-router-redux';
/* eslint-disable import/no-extraneous-dependencies */
import { createLogger } from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
/* eslint-disable import/no-extraneous-dependencies */

import reducers from './reducers';
import http from './common/http';
import queryParamsParserMiddleware from './middlewares/queryParamsParser';
import persistUserStateMiddleware, {
  reHydrateRootStateWithUser,
} from './middlewares/persistUserState';

export const thunkMiddleware = thunk.withExtraArgument(http);

export const history = createHistory();
const reduxRouterMiddleware = routerMiddleware(history);

const getMiddleware = () => {
  if (process.env.NODE_ENV === 'production') {
    return applyMiddleware(
      reduxRouterMiddleware,
      queryParamsParserMiddleware,
      persistUserStateMiddleware,
      thunkMiddleware
    );
  }
  return applyMiddleware(
    reduxRouterMiddleware,
    queryParamsParserMiddleware,
    persistUserStateMiddleware,
    thunkMiddleware,
    createLogger()
  );
};

export const store = createStore(
  reducers,
  reHydrateRootStateWithUser(),
  composeWithDevTools(getMiddleware())
);
