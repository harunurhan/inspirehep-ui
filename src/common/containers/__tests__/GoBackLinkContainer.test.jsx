import React from 'react';
import { shallow } from 'enzyme';
import { goBack } from 'react-router-redux';

import { getStore } from '../../../fixtures/store';
import GoBackLinkContainer, { dispatchToProps } from '../GoBackLinkContainer';

describe('GoBackLinkContainer', () => {
  it('render without props', () => {
    const wrapper = shallow(<GoBackLinkContainer store={getStore()} />).dive();
    expect(wrapper).toMatchSnapshot();
  });

  it('render with custom children', () => {
    const wrapper = shallow(
      <GoBackLinkContainer store={getStore()}>custom</GoBackLinkContainer>
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });

  it('calls dispatch with goBack() on click', () => {
    const mockDispatch = jest.fn();
    const { onClick } = dispatchToProps(mockDispatch);
    onClick();
    expect(mockDispatch).toHaveBeenCalledWith(goBack());
  });
});
