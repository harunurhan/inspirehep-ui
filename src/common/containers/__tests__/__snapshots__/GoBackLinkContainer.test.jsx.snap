// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoBackLinkContainer render with custom children 1`] = `
<LinkLikeButton
  onClick={[Function]}
>
  custom
</LinkLikeButton>
`;

exports[`GoBackLinkContainer render without props 1`] = `
<LinkLikeButton
  onClick={[Function]}
>
  go back
</LinkLikeButton>
`;
